tests/model_builder/test_model_builder.py

```diff
@@ -271,3 +274,27 @@ def test_id():
     ).hexdigest()[:16]
 
     assert model_builder.id == expected_id
+
+
+@pytest.mark.parametrize("name", ["prior_predictive", "posterior_predictive"])
+def test_sample_xxx_predictive_keeps_first(
+    fitted_model_instance, toy_X, name: str
+) -> None:
+    method_name = f"sample_{name}"
+    method = getattr(fitted_model_instance, method_name)
+
+    X_pred = toy_X["input"]
+
+    kwargs = {
+        "X_pred": X_pred,
+        "combined": False,
+        "extend_idata": True,
+        "random_seed": rng,
+    }
+    first_sample = method(**kwargs)
+    second_sample = method(**kwargs)
+
+    assert first_sample != second_sample
+
+    sample = getattr(fitted_model_instance.idata, name)
+    assert sample == first_sample
```

Comment:
Shouldn't it override the first with the second when `extend_idata` is True? 

Also we should parametrize this test to test both `extend_idata` cases

Reply:
Also I think these changes also apply to `fit`?

Reply:
This is the consequent from join="left"
I would think if you call the method again, you'd want to replace or be notified that it is being replaced. 

I think this is out of this PR's scope though\

Reply:
We should join right, otherwise extend_inferencedata=True makes little sense?

Reply:
That's what I meant to say. That's how I've change it. join="right"

Reply:
This test is still weird.

I expect the assert to be `assert sample == second_sample`.

The fact it is not failing tells me that assert is too premissive. I guess inferencedata equality is just based on groups or something. We need to check some actual values

Reply:
Okay. I will take a look later